src/wasm-builder.h

```diff
@@ -1336,7 +1285,9 @@ class Builder {
 
   // ensure a node is a block, if it isn't already, and optionally append to the
   // block
-  Block* blockify(Expression* any, Expression* append = nullptr) {
+  Block* blockify(Expression* any,
+                  Expression* append = nullptr,
+                  std::optional<Type> type = std::nullopt) {
```

Comment:
Oh neat, does this mean we can pass a type to the variadic version of `blockify` below now, too?

Reply:
I thought so, but it doesn't seem to work... It continues to `blockify(any, append)` and errors out as soon as the `append` is not `Expression*`, like this:
```console
/usr/local/google/home/aheejin/binaryen/src/wasm-builder.h:1307:12: error: no matc
hing member function for call to 'blockify'
 1307 |     return blockify(blockify(any, append), args...);
      |            ^~~~~~~~
/usr/local/google/home/aheejin/binaryen/src/passes/StackCheck.cpp:112:25: note: in
 instantiation of function template specialization 'wasm::Builder::blockify<wasm::
Type>' requested here
  112 |     auto *ret = builder.blockify(check, newSet, Type(Type::i32));
      |                         ^
/usr/local/google/home/aheejin/binaryen/src/wasm-builder.h:1288:10: note: candidat
e function not viable: no known conversion from 'wasm::Type' to 'Expression *' for
 2nd argument
 1288 |   Block* blockify(Expression* any,
      |          ^
 1289 |                   Expression* append = nullptr,
      |                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/google/home/aheejin/binaryen/src/wasm-builder.h:1306:10: note: candidat
e function template not viable: no known conversion from 'wasm::Type' to 'Expressi
on *' for 2nd argument
 1306 |   Block* blockify(Expression* any, Expression* append, Ts... args) {
      |          ^                         ~~~~~~~~~~~~~~~~~~
1 error generated.
```

I guess I should revert the `blockify` part. `blockifyWithName` should be fine given that it doesn't use a variadic parameter though.